v0/destinations/am/config.js

```diff
@@ -113,6 +113,11 @@ const BATCH_EVENT_ENDPOINT = "https://api.amplitude.com/batch";
 const GROUP_ENDPOINT = "https://api.amplitude.com/groupidentify";
 const ALIAS_ENDPOINT = "https://api.amplitude.com/usermap";
 
+const EU_ENDPOINT = "https://api.eu.amplitude.com/2/httpapi";
+const EU_BATCH_EVENT_ENDPOINT = "https://api.eu.amplitude.com/batch";
+const EU_ALIAS_ENDPOINT = "https://api.eu.amplitude.com/usermap";
+const EU_GROUP_ENDPOINT = "https://api.eu.amplitude.com/groupidentify";
```

Comment:
It's a bad idea to replicate all the constants for the EU server. It won't be extensible friendly. Instead, refactor these into a method.
I wouldn't insist on fixing it for the temporary release to the customer but should be addressed once eventually.
CC @utsabc

Reply:
agreed. It will be more efficient.

Reply:
agreed. It will be more efficient.

Reply:
With the changes introduced, we have increased our chances of making mistakes down the line with the API endpoints. 
Here is a better way to implement it,

```
const BASE_URL = "https://api.amplitude.com";
const BASE_URL_EU = "https://api.eu.amplitude.com";

const baseEndPoint = (destConfig) => {
  var retVal;
  switch (destConfig.residencyServer) {
    case "EU":
      retVal = BASE_URL_EU;
      break;
    default: // "US" or when it is not specified
      retVal = BASE_URL;
  }
  return retVal;
}

const defaultEndPoint = (destConfig) => {
  var retVal = `${baseEndPoint(destConfig)}/2/httpapi`;
  return retVal;
}

const identifyEndPoint = (destConfig) => {
  var retVal = `${baseEndPoint(destConfig)}/identify`;
  return retVal;
}

const batchEndPoint = (destConfig) => {
  var retVal = `${baseEndPoint(destConfig)}/batch`;
  return retVal;
}

const groupEndPoint = (destConfig) => {
  var retVal = `${baseEndPoint(destConfig)}/groupidentify`;
  return retVal;
}

const aliasEndPoint = (destConfig) => {
  var retVal = `${baseEndPoint(destConfig)}/usermap`;
  return retVal;
}
```

Note that I've also changed the config variable representation to the choice of the residency server.

Reply:
Nice implementation.
Updated as suggested.